Userland/Libraries/LibJS/Runtime/ECMAScriptFunctionObject.h

```diff
@@ -73,7 +73,7 @@ class ECMAScriptFunctionObject final : public FunctionObject {
     bool has_simple_parameter_list() const { return m_has_simple_parameter_list; }
 
     // Equivalent to absence of [[Construct]]
-    virtual bool has_constructor() const override { return !(m_is_arrow_function || m_kind == FunctionKind::Generator); }
+    virtual bool has_constructor() const override { return !(m_is_arrow_function || m_kind == FunctionKind::Generator || m_kind == FunctionKind::Async); }
```

Comment:
We can probably simplify this to `m_kind == FunctionKind::Regular && !m_is_arrow_function` since if async generators get added they won't be/have constructors either.

Reply:
fixed